### PR TITLE
deprecations fixes

### PR DESCRIPTION
**Pull request by raveit65**
@monsta @flexiondotorg @XRevan86 @lukefromdc @dnk 
Please test

### CODE REVIEW

**Comment by lukefromdc** [comment]
Works fine here with GTK3.21 Tested all 4 themes, they all work though 
the usual issue that "standard" gets light text when my theme is used 
remain. That is specific to my theme, which is normally used with "Coco"
anyway and no MATE theme has an issue with "standard."

File: src/themes/standard/theme.c [hunk 1]
```diff
@@ -699,14 +698,12 @@ GtkWindow* create_notification(UrlClickedCb url_clicked)
 	atk_object_set_description(atkobj, "Notification summary text.");
 
 	/* Add the close button */
-	alignment = gtk_alignment_new(1, 0, 0, 0);
-	gtk_widget_show(alignment);
-	gtk_box_pack_start(GTK_BOX(hbox), alignment, FALSE, FALSE, 0);
-
 	close_button = gtk_button_new();
+	gtk_widget_set_halign (close_button, GTK_ALIGN_END);
```
Comment (monsta) [inline src/themes/standard/theme.c]: Should we also set valign to "start" here to keep it the same as with gtk_alignment_new...?

File: src/themes/standard/theme.c [hunk 2]
```diff
@@ -699,14 +698,12 @@ GtkWindow* create_notification(UrlClickedCb url_clicked)
...
+	gtk_box_pack_start(GTK_BOX(hbox), close_button, FALSE, FALSE, 0);
 	windata->close_button = close_button;
 	gtk_widget_show(close_button);
-	gtk_container_add(GTK_CONTAINER(alignment), close_button);
+	gtk_container_add(GTK_CONTAINER(close_button), close_button);
```
Comment (monsta) [inline src/themes/standard/theme.c]: Wait... adding a widget into itself? :open_mouth:

File: src/themes/coco/coco-theme.c [hunk 3]
```diff
@@ -423,14 +421,11 @@ create_notification(UrlClickedCb url_clicked)
 
 	g_signal_connect (G_OBJECT (win), "composited-changed", G_CALLBACK (on_composited_changed), windata);
 
-    padding = gtk_alignment_new(0, 0, 0, 0);
-	gtk_widget_show(padding);
-	gtk_box_pack_start(GTK_BOX(main_vbox), padding, FALSE, FALSE, 0);
-  g_object_set(G_OBJECT(padding), "top-padding", 8, "right-padding", 8, NULL);
```
Comment (monsta) [inline src/themes/coco/coco-theme.c]: Should we add something (like setting margins) in place of this line?

File: src/themes/slider/theme.c [hunk 4]
```diff
@@ -713,19 +709,15 @@ void add_notification_action(GtkWindow* nw, const char* text, const char* key, A
...
 		gtk_widget_show(windata->actions_box);
 		update_content_hbox_visibility(windata);
 
-		alignment = gtk_alignment_new(1, 0.5, 0, 0);
-		gtk_widget_show(alignment);
-		gtk_box_pack_end(GTK_BOX(windata->actions_box), alignment, FALSE, TRUE, 0);
-
 		windata->pie_countdown = gtk_drawing_area_new();
 
+		gtk_widget_set_halign (windata->actions_box, GTK_ALIGN_END);
```
Comment (monsta) [inline src/themes/slider/theme.c]: This looks weird to me... actions_box is playing role of container two lines below, and pie_countdown is packed into it... so shouldn't halign be set on pie_countdown instead? Besides, halign on actions_box is set at line 465 already.

File: src/themes/slider/theme.c [hunk 5]
```diff
@@ -383,31 +382,30 @@ GtkWindow* create_notification(UrlClickedCb url_clicked)
 	gtk_widget_show(windata->main_hbox);
 	gtk_box_pack_start(GTK_BOX(main_vbox), windata->main_hbox, FALSE, FALSE, 0);
 
-	/* First row (icon, vbox, close) */
-	windata->iconbox = gtk_alignment_new(0.5, 0, 0, 0);
```
Comment (monsta) [inline src/themes/slider/theme.c]: Looks like iconbox variable isn't used anymore after this commit, so it can be removed.

**Comment by raveit65** [comment]
thanks, of course i will squash the fix commits after review

**Comment by lukefromdc** [comment]
One problem: the timeout pie is not appearing in any theme, not sure if that's intentional or not.

**Comment by monsta** [comment]
That's not good... pie countdown should be displayed.

**Comment by raveit65** [comment]
@monsta 
Have you ever test it?
The pie countdown displays only if a notification require it, you don't see that with with using mate-notification-properties.

**Comment by raveit65** [comment]
Btw. i don't know how to test it, notify-send don't have an option for it.

**Comment by monsta** [comment]
I know. I didn't test it yet. The only place I see it is network manager's notification about network being up.

I'll check it later...

**Comment by lukefromdc** [comment]
One way to test the pie is to make sure nm-applet is allowed to send 
notifications, then disable and re-enable networking.

**Comment by raveit65** [comment]
Branch is only rebased, no fix for the pie.
Confirmed, i broke the pie for standard, nodoka and coco themes.
With slider theme the pie seems already broken with 1.15.0, but i'm not sure if this really was working with gtk3.

**Comment by lukefromdc** [comment]
No pie in slider for a long time, that's not new. I never even knew it was
supposed to be supported there,

**Comment by lukefromdc** [comment]
Just tested today's code, works same as yesterday's version here.

**Comment by raveit65** [comment]
Hard logic issue without GtkAlignment \o/
But if i do the same for slider than the notification failed for nm-applet.
Here we have a problem with using GtkStyleContext for the pie.
For testing i replaced the pie code with pie code from standard theme, which use GtkStyle.
Et voila, i have a worse pie,........but i have a pie after removing the ! at the same place.
But sorry, fixing that SyleContext stuff is more than i able to do..

**Comment by lukefromdc** [comment]
Pies now work, all three themes that get the pie fix. In Coco it's exactly the same as before, don't recall how they were supposed to look in Nodoka or Standard since until now I'd only seen them on the preview from m-c-c.

**Comment by raveit65** [comment]
For standard theme i switched from ACTIVE to SELECTED color for the pie.
Which looks better imo.

**Comment by monsta** [comment]
Ok, pies are back, thanks :smile:
The new color in Standard theme is better visible indeed.

And yes, Slider indeed doesn't draw the pie, I noticed it before this PR.

The only thing that's not looking as before is the placement of the pie...

Before:





After:
1a
2a
3a

**Comment by raveit65** [comment]
You see everyerthing :smile: 
I just moved the pie to the left.
For me the design looks better, but i can revert this if you guys here prefer the old position.

**Comment by raveit65** [comment]
Btw, the button should not use colors from themes for coco and nodoka, imo.
Anyone an idea how to do that?

**Comment by monsta** [comment]
Ah... and I thought it was by mistake :smile:

So I've changed the packing of pie to ```gtk_box_pack_end``` and now the pie is in its usual place:

1b
2b
3b

The packing with ```gtk_box_pack_start``` might be better or worse, but I'd just prefer not to make UI changes when we're just fixing deprecations. We can always discuss the change later.

**Comment by monsta** [comment]
I'm not sure about using colors from themes, but I recall that Slider uses fg/bg from GTK+ theme... maybe it would help.

**Comment by raveit65** [comment]
That's the problem , for example coco use the colors from gtk theme for the button.
We should hard code it to a dark color here.

**Comment by monsta** [comment]
I agree, the notification theme should either use both fg/bg from GTK+ theme, or not use both of them at all. If we use only fg or only bg then we eventually stumble upon a GTK+ theme which makes the notification theme look bad :smile:

**Comment by monsta** [comment]
Ok, so I'll commit the change of pie position then?

**Comment by raveit65** [comment]
Btw. there are some FIXME for 'region' in standard theme under 'fil_background, i'm sure that is the cause of the weird borders.
Maybe you have an idea to fix that.

**Comment by raveit65** [comment]
Let me fix pie position, i want to rebase whole the PR i one single commit for every theme.

**Comment by raveit65** [comment]
OK, than i merge the PR

**Comment by monsta** [comment]
Ok

**Comment by raveit65** [comment]
merged
https://github.com/mate-desktop/mate-notification-daemon/commit/daca69be40ba1ea5e765a18b2a1185cafca2128c
https://github.com/mate-desktop/mate-notification-daemon/commit/264b9653dcba9f75a7385bc05a389e4252eefccb
https://github.com/mate-desktop/mate-notification-daemon/commit/5b8cecc43034d1668cce43a170e63eaa1a36459e
https://github.com/mate-desktop/mate-notification-daemon/commit/cbb5a9e9867b228611b9bb14cca62bdcccbd6dbf
https://github.com/mate-desktop/mate-notification-daemon/commit/7dc240a3fb0bec7d23dd00fea00890d8302b7f92
https://github.com/mate-desktop/mate-notification-daemon/commit/75965a8aa6c329974dec8d7a7d666c95a08d5b1f

**Comment by lukefromdc** [comment]
Using one and not the other will always leave some theme with invisible 
text. This is true on any widget in any program I suspect.

**Pull request closed without merging**